corehq/apps/reports/filters/users.py

```diff
@@ -331,22 +333,14 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs, request_user=None):
 
         q = user_es.UserES().domain(domain)
         if not can_access_all_locations:
-            # Make sure the passed in locations are accessible
-            location_ids = (SQLLocation.active_objects
-                            .get_locations(location_ids)
-                            .accessible_to_user(domain, request_user)
-                            .location_ids())
-            if HQUserType.REGISTERED in user_types:
-                all_accessible_loc_ids = (SQLLocation.active_objects
-                                          .accessible_to_user(domain, request_user)
-                                          .location_ids())
-                q = q.location(all_accessible_loc_ids)
-                return q.OR(
-                    filters.term("_id", user_ids),
-                    user_es.location(location_ids),
-                )
-            else:
-                return q.location(location_ids),
+            cls._verify_users_are_accessible(domain, request_user, user_ids)
+            return q.OR(
```

Comment:
how come this one is an OR, doesn't that mean that it will bypass the location restriction if the user's id is in user_ids?

Reply:
oh nevermind - i see that the user always has access to those users because of `_verify_users_are_accessible`